domain/notifier.go

```diff
@@ -1,10 +1,41 @@
 package domain
 
+import "time"
+
 type Notifier interface {
 	Notify([]Notification) error
 }
 
+type NotificationMessages struct {
+	ExpirationReminder   string `mapstructure:"expiration_reminder" default:"Your access to {{resource_name}} with role {{role}} will expire at {{expiration_date}}. Extend the access if it's still needed"`
+	AppealApproved       string `mapstructure:"appeal_approved" default:"Your appeal to {{resource_name}} with role {{role}} has been approved"`
+	AppealRejected       string `mapstructure:"appeal_rejected" default:"Your appeal to {{resource_name}} with role {{role}} has been rejected"`
+	AccessRevoked        string `mapstructure:"access_revoked" default:"Your access to {{resource_name}}} with role {{role}} has been revoked"`
+	ApproverNotification string `mapstructure:"approver_notification" default:"You have an appeal created by {{requestor}} requesting access to {{resource_name}} with role {{role}}. Appeal ID: {{appeal_id}}"`
+}
+
+const (
+	NotificationTypeExpirationReminder   = "ExpirationReminder"
+	NotificationTypeAppealApproved       = "AppealApproved"
+	NotificationTypeAppealRejected       = "AppealRejected"
+	NotificationTypeAccessRevoked        = "AccessRevoked"
+	NotificationTypeApproverNotification = "ApproverNotification"
+)
+
+type NotificationVariables struct {
```

Comment:
It'd be difficult to scale variables in a struct, should instead use a map interface and not have it here.
Parser method can validate if any of the variables is missing from the body when called.